app/gui2/shared/ast/tree.ts

```diff
@@ -54,6 +54,9 @@ export interface NodeMetadataFields {
   visualization?: VisualizationMetadata | undefined
   colorOverride?: string | undefined
 }
+function isNodeMetadataField(property: string) {
+  return ['position', 'visualization', 'colorOverride'].includes(property)
+}
```

Comment:
What about https://www.npmjs.com/package/ts-transformer-keys? You could use `keys<NodeMetadataFields>()`

Reply:
We can't use compiler plugins without switching from vite's default transpiler (esbuild) to tsc, which would have a build-time cost: https://github.com/evanw/esbuild/issues/1963.

We have a utility function `allKeys` that adds type-level correctness checking; I'll use that here. It is not as convenient as the TS-transformer (it requires an "example object" as an input), but it doesn't impose any unusual requirements on our build process.